mock/src/main/java/com/telefonica/mock/MockApiClient.kt

```diff
@@ -11,14 +11,31 @@ import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
 open class MockApiClient @Inject constructor(
-    private val dispatcher: CoroutineDispatcher,
+    private val coroutineDispatcher: CoroutineDispatcher,
     private val mockWebServer: MockWebServer,
 ) {
 
     private val mockMap: MutableMap<String, List<Mock>> = mutableMapOf()
 
+    private val enqueuedAnswers: MutableList<Mock> = mutableListOf()
+
+    private val dispatcher = object : Dispatcher() {
+        override fun dispatch(request: RecordedRequest): MockResponse {
+            val mockFiles: List<Mock> = enqueuedAnswers
```

Comment:
how will this code be executed for the mocked responses enqueued after invoking `setUp`?

Reply:
Oh, I see, this `dispatch` callback is executed on each request 👍